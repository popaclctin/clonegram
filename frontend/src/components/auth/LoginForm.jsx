import React from 'react';

function LoginForm() {
  return (
    <form>
      <div className='form-input'>
        <label htmlFor='email'>Email</label>
        <input type='email' id='email' required />
      </div>
      <div className='form-input'>
        <label htmlFor='password'>Password</label>
        <input type='password' id='password' required />
      </div>
      <button type='submit'>Login</button>
    </form>
  );
}

export default LoginForm;
