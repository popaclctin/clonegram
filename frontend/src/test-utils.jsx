import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { setupStore } from './store';
import { setupListeners } from '@reduxjs/toolkit/dist/query';

import { BrowserRouter } from 'react-router-dom';

export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  } = {}
) {
  setupListeners(store.dispatch);

  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    );
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
