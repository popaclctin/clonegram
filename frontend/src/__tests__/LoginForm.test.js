import React from 'react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { fireEvent, screen } from '@testing-library/react';
import { renderWithProviders } from '../test-utils';
import LoginForm from '../components/auth/LoginForm';

export const handlers = [
  rest.get('/api/user', (req, res, ctx) => {
    return res(ctx.json('John Smith'), ctx.delay(150));
  }),
];

const server = setupServer(...handlers);

// Enable API mocking before tests.
beforeAll(() => server.listen());

// Reset any runtime request handlers we may add during the tests.
afterEach(() => server.resetHandlers());

// Disable API mocking after the tests are done.
afterAll(() => server.close());

it('should take a snapshot', () => {
  const { asFragment } = renderWithProviders(<LoginForm />);
  expect(asFragment(<LoginForm />)).toMatchSnapshot();
});
