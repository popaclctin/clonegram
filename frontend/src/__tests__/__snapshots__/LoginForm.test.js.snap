// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <section
    class="authForm"
  >
    <h1
      class="authForm__title"
    >
      Clonegram
    </h1>
    <form>
      <input
        class="authForm__field"
        id="email"
        name="email"
        placeholder="Email"
        type="email"
        value="bula@gmail.com"
      />
      <input
        class="authForm__field"
        id="password"
        name="password"
        placeholder="Password"
        type="password"
        value="parola"
      />
      <button
        class="authForm__submitBtn"
        type="submit"
      >
        Log In
      </button>
    </form>
    <a
      class="authForm__forgotPass"
      href="#"
    >
      Forgotten your password?
    </a>
  </section>
</DocumentFragment>
`;
